### PR TITLE
cmake changes for windows compiling

### PR DESCRIPTION
**Pull request by shz0116**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -152,8 +152,23 @@ set_target_properties(fbgemm_generic fbgemm_avx2 fbgemm_avx512 PROPERTIES
...
+        CMAKE_CXX_FLAGS_MINSIZEREL CMAKE_CXX_FLAGS_RELWITHDEBINFO)
+       if(${flag_var} MATCHES "/MD")
+         string(REGEX REPLACE "/MD" "/MT" ${flag_var} "${${flag_var}}")
+       endif(${flag_var} MATCHES "/MD")
+     endforeach(flag_var)
+  endif()
   target_compile_options(fbgemm_avx2 PRIVATE "/arch:AVX2")
   target_compile_options(fbgemm_avx512 PRIVATE "/arch:AVX512")
+
+  set(CMAKE_EXE_LINKER_FLAGS    "${CMAKE_EXE_LINKER_FLAGS} /NODEFAULTLIB:MSVCRT /ignore:4049")
```
Comment (jspark1105) [inline CMakeLists.txt]: What's 4049 warning? Can we add a comment on it?

File: test/CMakeLists.txt [hunk 2]
```diff
@@ -18,8 +24,15 @@ macro(add_gtest TESTNAME)
   set_target_properties(${TESTNAME} PROPERTIES
           CXX_STANDARD 11
           CXX_EXTENSIONS NO)
- target_compile_options(${TESTNAME} PRIVATE
-   "-m64" "-mavx2" "-mfma" "-masm=intel")
+  if(MSVC)
+    target_compile_options(${TESTNAME} PRIVATE "/arch:AVX2")
```
Comment (jspark1105) [inline test/CMakeLists.txt]: In MSVC, AVX2 implies FMA?

File: CMakeLists.txt [hunk 3]
```diff
@@ -152,8 +152,23 @@ set_target_properties(fbgemm_generic fbgemm_avx2 fbgemm_avx512 PROPERTIES
...
 
 if(MSVC)
+  if(FBGEMM_LIBRARY_TYPE STREQUAL "static")
+     target_compile_definitions(fbgemm_generic PRIVATE ASMJIT_STATIC)
+     target_compile_definitions(fbgemm_avx2 PRIVATE ASMJIT_STATIC)
+     target_compile_definitions(fbgemm_avx512 PRIVATE ASMJIT_STATIC)
+     foreach(flag_var
+        CMAKE_CXX_FLAGS CMAKE_CXX_FLAGS_DEBUG CMAKE_CXX_FLAGS_RELEASE
+        CMAKE_CXX_FLAGS_MINSIZEREL CMAKE_CXX_FLAGS_RELWITHDEBINFO)
+       if(${flag_var} MATCHES "/MD")
```
Comment (jspark1105) [inline CMakeLists.txt]: Can we add a comment on what we're doing?

File: CMakeLists.txt (hunk 1 above)
Comment (shz0116) [reply CMakeLists.txt]: I will add comments for these changes.
We can ignore this one. I forgot to remove it.

File: test/CMakeLists.txt (hunk 2 above)
Comment (shz0116) [reply test/CMakeLists.txt]: I think so. But need further check.

**Pull request closed without merging**